src/devices/gridstream.c

```diff
@@ -0,0 +1,258 @@
+                sprintf(srcaddress_str, "%02x%02x%02x%02x", b[26], b[27], b[28], b[29]);
+                if (stream_len == 0x47) {
+                    clock  = ((uint32_t)b[16] << 24) | (b[17] << 16) | (b[18] << 8) | b[19];
+                    uptime = ((uint32_t)b[24] << 24) | (b[25] << 16) | (b[26] << 8) | b[27];
+                    sprintf(srcwanaddress_str, "%02x%02x%02x%02x%02x%02x", b[32], b[33], b[34], b[35], b[36], b[37]);
+                    strftime(clock_str, sizeof(clock_str), "%a %Y-%m-%d %H:%M:%S %Z", localtime(&clock));
+
+                    /* clang-format off */
+                            data = data_make(
+                                    "model",        "", DATA_STRING, "LandisGyr GridStream",
+                                    "id",           "Source Meter ID", DATA_STRING, srcaddress_str,
+                                    "subtype",      "", DATA_INT,    subtype,
+                                    "destaddress",  "Target Meter ID", DATA_STRING, destaddress_str,
+                                    "timestamp",    "", DATA_STRING, clock_str,
+                                    "uptime",       "", DATA_INT,    uptime,
+                                    "wanaddress",   "", DATA_STRING, srcwanaddress_str,
+                                    "mic",          "", DATA_STRING, "CRC", // CRC, CHECKSUM, or PARITY
+                                    NULL);
+                    /* clang-format on */
+                }
+                else {
+
+                    /* clang-format off */
+                            data = data_make(
+                                    "model",        "", DATA_STRING, "LandisGyr GridStream",
+                                    "id",           "Source Meter ID", DATA_STRING, srcaddress_str,
+                                    "subtype",      "", DATA_INT,    subtype,
+                                    "destaddress",  "Target Meter ID", DATA_STRING, destaddress_str,
+                                    "mic",          "Integrity", DATA_STRING, "CRC", // CRC, CHECKSUM, or PARITY
+                                    NULL);
+                    /* clang-format on */
+                }
+                break;
+            }
+            break;
+        default:
+            return DECODE_ABORT_LENGTH;
+            break;
+        }
+
+        decoder_output_data(decoder, data);
+
+        if (decoder->verbose >= 2) {
+            decoder_log_bitrow(decoder, 2, __func__, b, decoded_len * 8, "frame data");
+        }
+        // Return 1 if message successfully decoded
+        return 1;
+    }
+    else {
+        return DECODE_FAIL_SANITY;
+    }
+}
+
+static char const *const output_fields[] = {
+        "model",
+        "id",
+        "subtype",
+        "wanaddress",
+        "destaddress",
+        "uptime",
+        "srclocation",
+        "destlocation",
+        "timestamp",
+        "mic",
+        NULL,
+};
+
+r_device const gridstream = {
```

Comment:
Duplicate this for the other rates and change the name to match the rate.

Reply:
Added the other rates, and they appear to work, at least with my saved samples. However, I cannot determine a way to output the bitrate in the data table.